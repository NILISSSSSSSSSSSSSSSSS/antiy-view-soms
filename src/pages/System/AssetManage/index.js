import { Component } from 'react'
import { connect } from 'dva'
import { Form } from 'antd'
import hasAuth from '@/utils/auth'
// import './style.less'
// import hasAuth from '@/utils/auth'
// import * as regExp from '@/utils/regExp'
import Subfield from '@/components/System/Subfield'
import { systemPermission } from '@a/permission'
// import AddTreeModal from '@/components/System/AreaTreeModal'
class SystemAsset extends Component {
  constructor (props) {
    super(props)
    this.state = {
      treeData: this.props.treeData,
      elList: []
    }
  }

  componentDidMount () {
    //获取管理区域树数据
    // this.getTree()
  }

  UNSAFE_componentWillReceiveProps (nextProps) {
    if (JSON.stringify(this.props.treeData) !== JSON.stringify(nextProps.treeData)) {
      this.setState({
        treeData: nextProps.treeData
      })
    }
  }

  authority = () => {
    const auth = ['edit', 'add', 'delete', 'save']
    const init = [systemPermission.sysAreaEdit, systemPermission.sysAreaAdd, systemPermission.sysAreaDelete, systemPermission.sysAreaSave].map((item, i) => {
      return hasAuth(item) ? auth[i] : undefined
    }).filter(item => item !== undefined)
    return init
  }
  render () {
    const { treeData, elList } = this.state
    return (
      <div className="main-table-content common-tree">
        <div className="system-Asset">
          <Subfield
            iData={treeData}
            title='区域'
            handle={['updateSingle', 'saveSingle', 'deleteSingle']}
            apiConfig={{
              api: 'getTree',
              config: {
                userId: sessionStorage.getItem('id')
              }
            }}
            authList={this.authority()}></Subfield>
          {
            elList
          }
        </div>
      </div>
    )
  }
}

const mapStateToProps = ({ system }) => {
  return { treeData: system.treeData }
}

const SystemAssetForm = Form.create()(SystemAsset)

export default connect(mapStateToProps)(SystemAssetForm)
