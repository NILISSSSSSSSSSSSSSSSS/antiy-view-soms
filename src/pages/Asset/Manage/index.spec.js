import React from 'react'
import { shallow } from 'enzyme'
import AssetManage from './index'

const props = {
  location: {
    search: '?status=1'
  },
  history: {
    push: jest.fn()
  }
}
jest.mock('@/utils/auth', ()=>{
  return jest.fn((auth)=>{
    const { assetsPermission } = require('@a/permission')
    return  [ assetsPermission.ASSET_INFO_LIST, assetsPermission.ASSET_UNKNOWN_INFO ].includes(auth)
  })
})
describe('资产信息管理 <AssetManage />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<AssetManage { ...props}/>)
  })
  it('快照', () => {
    const app = wrapper.debug()
    expect(app).toMatchSnapshot()
  })
  it('资产详情的正常渲染', () => {
    expect(wrapper.find('.main-table-content')).toHaveLength(1)
  })
  it('tabChange ==> 切换tabs', () => {
    const instance = wrapper.instance()
    const spy_tabChange = jest.spyOn(instance, 'tabChange')
    spy_tabChange('1')
    spy_tabChange('2')
  })
})
