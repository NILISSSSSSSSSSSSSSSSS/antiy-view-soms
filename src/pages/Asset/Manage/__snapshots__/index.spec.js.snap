// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`资产信息管理 <AssetManage /> 快照 1`] = `
"<div className=\\"main-table-content\\">
  <Tabs activeKey=\\"1\\" onChange={[Function]} hideAdd={false} tabPosition=\\"top\\">
    <TabPane tab=\\"资产信息列表\\" forceRender={true} placeholder={{...}}>
      <withRouter(Form(Connect(Information)))>
        [function children]
      </withRouter(Form(Connect(Information)))>
    </TabPane>
    <TabPane tab=\\"未知资产\\" forceRender={true} placeholder={{...}}>
      <withRouter(Unknown)>
        [function children]
      </withRouter(Unknown)>
    </TabPane>
  </Tabs>
</div>"
`;
